commit/merkleroot/outcome.go

```diff
@@ -46,9 +46,14 @@ func (w *Processor) getOutcome(
 
 	switch nextState {
 	case SelectingRangesForReport:
-		return reportRangesOutcome(commitQuery, consensusObservation), nextState
+		return reportRangesOutcome(q, consensusObservation), nextState
 	case BuildingReport:
-		return buildReport(commitQuery, consensusObservation, previousOutcome), nextState
+		if q.RetryRMNSignatures {
+			// We want to retry getting the RMN signatures on the exact same outcome we had before.
+			// The current observations should all be empty.
```

Comment:
Does it make sense to assert this at all? Maybe in `ValidateObservation`, if `q.Retry` is true, the observation should be empty?

Reply:
Makes sense, yeah